### DISCUSSION TITLE
Adding HTML DIV Element With Specific Id

### DISCUSSION THREAD

**Discussion by omerharuncetin** [Milkdown/milkdown#776] [Q&A]
Hello,

I am trying to embed a youtube iframe into the editor. The process will work as follows:

1. User enters a link ✔️ 
2. If the link includes a youtube video id, I will parse it and get the video id ✔️ 
3. I need to create a `div` with a specific id. ❌ 
4. I need to add that `div` into the editor. ❌ 

Can anyone please help me how to add a custom div to milkdown editor?

**Details:**

- I am using the editor for Markdown to HTML
- Milkdown version is 6.3.0

I am looking forward to your replies, it is an urgent situation. Thanks in advance.

**Comment by Saul-Mirone** [comment]
You need to define a new type of `node` to do that. You'll need to consider following steps:

1. You need to create a remark plugin, which will create a new type of node, let's call it `YouTubeNode`.
2. You need to create a new type of `Node`, you should define the schema for it. Render the node as youtube iframe, and encode it into a link when serializing.
3. Add an inputrule to transform the user input into your `YouTubeNode`

**Reply by omerharuncetin** [reply]
Thank you so much. I used the iframe plugin for that and it is enough for me.

I want to ask another question, how can I create a command for the Iframe plugin at milkdown.dev ?

**Reply by Saul-Mirone** [reply]
What will the command do?
